python/src/energuide/snippets.py

```diff
@@ -2,119 +2,74 @@
-    windows = house.findall('Components/*/Components/Window')
+def _extract_values(node: etree._Element,
+                    xpath_mapping: typing.Dict[str, str]) -> typing.Dict[str, typing.Optional[str]]:
+    output = {key: node.xpath(value) for key, value in xpath_mapping.items()}
+    return {key: value[0] if value else None for key, value in output.items()}
+
+
+def _ceiling_snippet(ceiling: etree._Element) -> typing.Dict[str, typing.Optional[str]]:
+    return _extract_values(ceiling, {
+        'label': 'Label/text()',
+        'typeEnglish': 'Construction/Type/English/text()',
+        'typeFrench': 'Construction/Type/French/text()',
+        'nominalRsi': 'Construction/CeilingType/@nominalInsulation',
+        'effectiveRsi': 'Construction/CeilingType/@rValue',
+        'area': 'Measurements/@area',
+        'length': 'Measurements/@length',
+    })
+
+
+def _floor_snippet(floor: etree.ElementTree) -> typing.Dict[str, typing.Optional[str]]:
+    return _extract_values(floor, {
+        'label': 'Label/text()',
+        'nominalRsi': 'Construction/Type/@nominalInsulation',
+        'effectiveRsi': 'Construction/Type/@rValue',
+        'area': 'Measurements/@area',
+        'length': 'Measurements/@length',
+    })
+
+
+def _wall_snippet(wall: etree.ElementTree) -> typing.Dict[str, typing.Optional[str]]:
+    return _extract_values(wall, {
+        'label': 'Label/text()',
+        'constructionTypeCode': 'Construction/Type/@idref',
+        'constructionTypeValue': 'Construction/Type/text()',
+        'nominalRsi': 'Construction/Type/@nominalInsulation',
+        'effectiveRsi': 'Construction/Type/@rValue',
+        'perimeter': 'Measurements/@perimeter',
+        'height': 'Measurements/@height',
+    })
+
+
+def _door_snippet(door: etree.ElementTree) -> typing.Dict[str, typing.Optional[str]]:
+    return _extract_values(door, {
+        'label': 'Label/text()',
+        'typeEnglish': 'Construction/Type/English/text()',
+        'typeFrench': 'Construction/Type/French/text()',
+        'rsi': 'Construction/Type/@value',
+        'height': 'Measurements/@height',
+        'width': 'Measurements/@width',
+    })
+
+
+def _window_snippet(window: etree.ElementTree) -> typing.Dict[str, typing.Optional[str]]:
+    return _extract_values(window, {
+        'label': 'Label/text()',
+        'constructionTypeCode': 'Construction/Type/@idref',
+        'constructionTypeValue': 'Construction/Type/text()',
+        'rsi': 'Construction/Type/@rValue',
+        'width': 'Measurements/@width',
+        'height': 'Measurements/@height',
+    })
+
+
+def snip_house(house: etree.ElementTree) -> typing.Dict[str, typing.List[typing.Dict[str, typing.Optional[str]]]]:
+    ceilings = house.xpath('Components/Ceiling')
+    floors = house.xpath('Components/Floor')
+    walls = house.xpath('Components/Wall')
+    doors = house.xpath('Components/Wall/Components/Door')
+    windows = house.xpath('Components/*/Components/Window')
```

Comment:
I believe that these two need to be updated to the change in the search path introduced in #104

Reply:
Thank you - that was the source of the merge conflict and I didn't update them properly.